### DISCUSSION TITLE
GSoC 2023: Contributor Proposal Review and Updates

### DISCUSSION THREAD

**Discussion by nate-double-u** [cncf/mentoring#918] [Google Summer of Code 2023]
Hello everyone, 

The contributor proposal submission has now closed, and we're now moving into the proposal review phase.

Currently there is a bug in the system where Ignoring an individual proposal will ignore it for the entire org. GSoC is working on this, and we'll be reaching out to the team to make sure that all the mentors are able to see all the proposals that are intended for them.

I'm including the GSoC Program Administrators' email in this post to help explain the situation, and to provide information on next steps.

Cheers,
Nate.

---

From: summerofcode-noreply@google.com
Subject: GSoC 2023: Contributor Proposal Review and Updates
Sent: Tue, Apr 4, 2023 at 9:35 PM PT

---

Hello,

The GSoC 2023 contributor proposal submission period has now closed. We had **5,704 contributors submit 7,723 proposals** for GSoC 2023. These are some of the highest numbers we’ve had for GSoC. Thank you all for taking the time to work with prospective contributors during the application period (and before) to help them understand more about your org and the project ideas you were interested in having GSoC contributors work on this summer.

In this phase of GSoC (Proposal Review and Ranking), your org will look through all the contributor proposals you received and choose which ones your org wishes to select as GSoC 2023 contributor projects. You will also be determining which mentors will be assigned to each contributor project so your org admins can rank the proposals before April 27.

## Updates

1. **IGNORE Proposal Change**: The ignore feature was not implemented the way we thought it had been implemented. The way the feature actually works is that any Org Admin or Mentor can mark a proposal as ‘Ignored’ and it will go into the ‘Ignored Proposals’ for your Org (for everyone).
    1. In some ways this is handy (and what some folks wanted) as when something is truly ‘spammy’ and should be ignored one person can see that and then Ignore the proposal saving everyone else the headache of even looking at it.
    1. But, if someone is ignoring proposals just because they don’t want to mentor them, that is not the way this feature should be used. Folks should use the ‘Starred’ feature as their way to determine the proposals they care about (that feature is in fact user specific and only you can see what you star (or don’t star).
1. **CSV Export of Proposals doesn’t give you PDF link**.The csv is not pulling the correct signed URL so that pdf link in the export doesn’t have the correct proposal ID. Developers are working on this and the fix should be in late Wed/Thursday. 3. In the meantime you can still view proposals through the dashboard one by one (and star for yourself or ignore for the whole org, etc.). 4. We will notify you as soon as the CSV is pulling the pdf link correctly.
1. **People missing the 1800 UTC deadline**. 254 submitted proposals in the final 10 minutes, with 38 people registering in the final 10 minutes. Now 95% of those are probably spammy/not quality proposals (unless they were revisions of a previous proposal). Your org may receive an email from someone who is upset they missed the deadline by a few minutes, didn’t realize 1800 UTC didn’t mean 1800 their local time, or they didn’t submit their proposal through the GSoC website. All of these issues are the user’s responsibility and we don’t make exceptions and extend deadlines.

### Reviewing Contributor Proposals: April 4 - 27

Mentors and Org Admins can view your organization's Contributor Proposals on your dashboard's Inbox tab.

**Interested Mentors count** How many Mentors have shown interest in mentoring a given proposal. Only OAs can see the actual Mentor display names.

**Project Size** The project size the Contributor indicated for their project. The size cannot be changed by the Org Admin (it wouldn’t be right for an OA to be able to make a project bigger or smaller than what the contributor proposed). The size the contributor indicates on their proposal should match the estimated amount of work in the proposal.

**Select to Rank** Viewable by Mentors and OAs. Only OAs can select a proposal to rank, but both can see if a proposal has chosen to go into the ranking queue for possible ranking. Ranking is now available (and closes April 27th 1800 UTC).

### Proposal Disappeared?

You may have been chatting with contributors these past couple of weeks but you no longer see their proposal in the system for you to select, that is because either 1) they withdrew themselves from the program 2) they removed their proposal in favor of submitting a different one to your org or another org or 3) someone in your org moved their proposal to the ignored proposals (you can view ignored proposals and ‘un-ignore’ if you feel strongly about that proposal being considered).

**Your decision about whether or not to select a contributor should be based on the proposal and your interactions with them over the past month+.**

### Contributor Eligibility

Contributors have registered and agreed to the Program Rules attesting that they are students or beginners to open source coding. As mentors and OAs, you are choosing a student or beginner contributor that is not only best suited for your project and organization, but who would benefit from the mentorship you are providing. An applicant that is an experienced software engineering professional is not the intended GSoC 2023 contributor. As you review the proposals and your previous interactions with interested contributors, here is the FAQ answer to [what constitutes an open source beginner](https://developers.google.com/open-source/gsoc/faq#how_do_i_know_if_i_am_considered_a_beginner_in_open_source_development) for reference.

Orgs can not accept contributors who are immediate family members (parent, sibling, child, spouse, or life partner) or a member of the same household (whether related or not) of any Org Admin or Mentor for your organization per the Program Rules.

#### Want to Mentor?

If you are interested in being a mentor for a given proposal then you can click the “Want to Mentor” button on the top right corner on the proposal’s page to help your Org Admin know that you are interested. This is particularly helpful for the Org Admins when determining how many people are excited about a proposal and how they want to rank the proposal.

We suggest projects have 2, and even 3, mentors ready to help mentor each contributor project. 1 mentor should be considered the primary mentor (responsible for evaluations, etc.) and the other 1-2 mentors should help primary mentors as much as they can and be ready to step in should the primary mentor need to take some time off for any period of time at any point in the program.

#### Can I mentor multiple students?

The general advice from other mentors is, it’s best to mentor 1 contributor and focus on that contributor so you can give them your full attention, but it’s okay to be a backup mentor for a second project. Don’t spread yourself too thin or it will be a bad experience for everyone - you, the contributor, and the org.

There isn’t a rule against mentoring multiple contributors but if this is your first time mentoring we strongly recommend you mentor 1 contributor. You will see many discussions about this on the mentor list archives.

#### As the primary mentor, how much time will I spend mentoring a contributor?

This all depends on the project, contributor and how you both decide to structure the project coding period - will this be a standard 12 week project or something longer (up to 22 weeks) or as short as 10 weeks? Time commitments can vary.

Generally you can assume that during the community bonding period you will want to commit 5+ hours a week to help set the contributor up for success and integrate them into your community. Once coding begins, you should be prepared to spend 2.5 - 4 hours per week with the contributor reviewing their work, giving feedback, answering questions, having video chats, etc.

### Quick Glossary

**Proposal**: A document describing the project the contributor wants to work on for your organization. It should include a timeline with clear milestones and specific deliverables.

**Ranked Proposals**: Org Admins will rank their org’s selected proposals to prioritize which proposals they would like to accept. If an org ranks 10 proposals they are asking Google for 10 project slots.

**Project slot**: Google Admins will grant each org a certain number of project “slots” based on requests, history with GSoC and their number of available mentors. The number of project slots granted to an org will translate to the top ranked proposals for that org (in order). 7 project slots means ranked slots #1 - #7 will become that org’s Contributor Projects.

**Contributor Project**: A selected Contributor’s Proposal that is among the org’s granted, ranked project slots.

**Project Length**: Project length means a 10, 12, 14, 16, 18, 20 or 22 week project. A project’s size (175hr or 350hr) is in no way related to the project length. A 175hr-medium size project can be 10 or up to 22 weeks in length. And likewise a 350hr-large size project can be anywhere between 10 and 22 weeks (in 2 week increments of course).
Proposals Ranked - deadline Thursday, April 27 18:00 UTC (Org Admin responsibility)

All Org Admins should read [Project Selection](https://developers.google.com/open-source/gsoc/help/slot-allocation) for details.

* Org Admins will rank their selected proposals based on discussions with their community members.
* Org Admins will prioritize the projects most important to their org by ranking them the highest.
* If an OA ranks 10 proposals (#1 - #10) they are asking Google for 10 project slots. If Google grants them 8 projects slots the org will automatically receive ranked projects #1 - #8.
* If 2 orgs select the same contributor, the org that ranked the contributor’s proposal the highest will receive that contributor.
  * The other org will receive the next ranked proposal in their list (if they requested 15 slots and received 12 but #8 was a contributor that had a ranked proposal higher with another org, the system will automatically accept #13 for the org). If an org received all of their requested slots but one contributor slot was ranked higher by another org they will be contacted by Google Admins with an option to fill that spot with another proposal.
* Mentors are assigned when the OA ranks a proposal. At least 1 mentor must be selected (more can be added later at anytime).
* Do not accept "okay" proposals.
* New orgs will likely only receive 1-2 project slots, no matter how many they request.
* Be sure you have your mentors lined up before the Proposal ranking deadline. If an Org requests 10 slots and only has 5 mentors that have registered for the 2023 program, Google Admins will likely only give the org 5 slots or less. Your org needs to demonstrate you have plenty of mentors available to mentor your GSoC contributors.

### Viewing Proposals

Mentors and Org Admins will be able to see your organization's Contributor Proposals on your dashboard's Inbox tab.

* Stars - (visible only to you) Can be used to help you keep track of your favorite proposals. You can also filter by just your starred or unstarred proposals.
* Proposal title - Click to view Proposal Details.
* Submitted Date - When the Contributor first submitted their proposal
* Last Updated - Will help you see if they have made more changes since the first submitted their proposal
* Interested Mentors count - How many Mentors have shown interest in mentoring a given proposal. Only OAs can see the actual Mentor display names.
* Project Size - The project size the Contributor indicated for their project. If the project size is not what you were expecting, be sure to contact the contributor to discuss before the Contributor deadline April 4th. The size cannot be changed after the deadline. The size the contributor indicates on their proposal should match the estimated amount of work in the proposal.
* Selected to Ignore - Viewable by Mentors and OAs. Any proposals you may choose to ignore will still be visible by others.
* Select to Rank- Viewable by Mentors and OAs. Only OAs can select a proposal to rank, but both can see if a proposal has been selected to go into the ranking queue. Ranking begins when Contributor Apps close on April 4 at 1800 UTC.

On the Proposal Details Page, you can view all the details, including the contributor's email address and the full proposal PDF. You can also indicate (to your OA) your interest in mentoring by clicking the 'Want to Mentor' button.

### Ignoring Proposals

Many orgs receive dozens or even hundreds of proposals, some of those are what would be considered spammy. For example, some people will submit a proposal to 3 different orgs but it’s clear the proposal is not actually targeted to the work the org does, it’s just them submitting anything thinking it could get them accepted. Also if you find a proposal that appears to have been copied from a previous participant that would certainly be considered spammy (and you can report these to Google Admins as we remove applicants who are copying other people’s proposals, a violation of the Program Rules they agreed to. You and your community don’t want to look at those time and time again so now there is a way to ignore those proposals.

When Mentors and Org Admins are in the Manage Proposals - > Inbox on their GSoC dashboard they can go to the Selected to ignore column. When you click the ignore icon for a proposal the icon will turn red and will disappear from your screen (and from view for all of your 2023 Org Members) as it is sent to the ‘Ignored’ proposals.

The filtered view of Proposals contains 4 options in the dropdown box labeled ‘Starred’:

* All Proposals (the default view and means essentially anything not ignored and these can be selected to be ranked)
* Starred (your personal view)
* Un-starred (your personal view)
* Ignored (all ignored proposals for your org, these can not be selected to be ranked; you can move a proposal from the ignored list by unclicking the circle with the line through it icon - it then will go into the ‘All Proposals’ view that can be selected to be ranked)

The Ignored Proposals should only be used for spammy proposals - not because you don’t like the project. Another mentor may like the project but if you ignore it then it is removed from the All Proposals view and the ability to be ranked from the entire org (until someone else ‘un-ignores’ the proposal which will put it back into the All Proposals view of proposals than can be selected to be Ranked by Org Admins).

### Changing Project Length from Standard 12 week project

On May 3rd when Google announces the Contributor projects to the orgs, by default all projects will have the 12 week project length. At that point OAs will be able to go into the project details for each Contributor project and adjust the timeline from a minimum of 10 weeks to a maximum of 22 weeks. OAs are encouraged to have these edits wrapped up by the end of the community bonding period. We lay out all of the project deadlines for you at the top of each project - you MUST use the dates we provide, you can’t randomly make up a deadline that is not in the dates we supply.

OAs can still make timing adjustments throughout the coding period as contributors have things come up and they need to adjust their timelines, but having the general dates locked down by the start of the coding period will be a better experience for mentors and contributors.

### Code of Conduct

Be sure to read the [Program Rules](https://summerofcode.withgoogle.com/rules/) and the [Code of Conduct](https://opensource.google.com/conduct/) section for GSoC. Your org may also have an additional Code of Conduct which you should be sure to point students to as well. If you have a situation where a prospective student, mentor or org admin violates the GSoC Code of Conduct please notify me as soon as possible at [sttaylor@google.com](mailto:sttaylor@google.com).
Important Upcoming Dates

Please be sure to mark these dates in your calendar of choice.

Use this [link](https://calendar.google.com/calendar/embed?src=gsummerofcode%40gmail.com&ctz=America%2FLos_Angeles) to add the Google Summer of Code dates to your Google Calendar - remember Evaluation deadlines will vary depending on your contributor’s project length. _We are only stating the midterm and final dates for the standard 12 week project in the calendar and below - if your contributor project is a different length be sure to mark your own calendar accordingly._

**April 4 - 26**: Review all submitted GSoC Contributor proposals with your org and consider how many you want to select and how many you can handle (how many you have committed mentors for). Decide on the ranking of your GSoC Contributor slots. At least 1 mentor must be assigned to each project before it can be selected.

**April 27 18:00 UTC**: Deadline to submit ranked slot requests (Org Admins enter requests)

**April 27 - May 2**: Google Program Admins review and assign org slots

**May 3**: Organizations receive notification of their accepted GSoC 2023 Contributors

**May 4**: Accepted GSoC 2023 GSoC Contributor projects are announced

**May 4 - 28**: Community Bonding Period

**May 27**: Deadline to notify Google Admins of an inactive GSoC Contributor that you wish to remove from the program

**May 29**: Coding begins

Dates for Midterms and Final Evaluations will vary depending on Project length

**September 6**: Initial batch of successful GSoC Contributors passing GSoC 2023 are announced

**October 13-15**: In person GSoC 2023 Mentor Summit (Sunnyvale, CA) Details in April

### Receiving Emails from Google Program Administrators

To avoid emails from us possibly being marked as spam please add [summerofcode-noreply@google.com](mailto:summerofcode-noreply@google.com) and [gsoc-support@google.com](mailto:gsoc-support@google.com) as trusted contacts.

### Email Archive

We have created a [GSoC 2023 Mentor and Org Admin email archive](https://docs.google.com/document/d/1K3J323__RkqGJGFmFCQ-BJMt-S9Ct0cRojD2KgMlpAk/edit?pli=1#) where you and all mentors and org admins in this year’s program will be able to reference all of the important emails that are being sent over the next 7 months. We suggest you encourage any new mentors or org admins to review the archive when they join your org along with the [Mentor Guide](https://google.github.io/gsocguides/mentor/).

### Important Links

* [Project Selection](https://developers.google.com/open-source/gsoc/help/slot-allocation)
* [Mentor Guide](https://google.github.io/gsocguides/mentor/)
* [Being a Great GSoC Mentor video](https://www.youtube.com/watch?v=3J_eBuYxcyg)
* [Org Admin tips](https://developers.google.com/open-source/gsoc/help/oa-tips)
* [Roles and Responsibilities](https://developers.google.com/open-source/gsoc/help/responsibilities)
* [GSoC 2023 Mentor and Org Admin email archive](https://docs.google.com/document/d/1K3J323__RkqGJGFmFCQ-BJMt-S9Ct0cRojD2KgMlpAk/edit?pli=1#)

### Thank you

We appreciate your patience and feedback with these annoying UI and general webapp issues. We have vendors that build our webapp and this year we have limited resources for the webapp as we put more funds toward being able to accept as many GSoC contributors as possible for 2023 so there are a lot of things on our wish list but we have to prioritize constantly to try and get the biggest wins for y’all as possible. Thank you for your understanding.

### Contact Us

If you have any questions please contact us at [gsoc-support@google.com](mailto:gsoc-support@google.com)

Thank you for being a part of Google Summer of Code 2023, this program could not happen without all of you!

Best,

Stephanie Taylor, Perry Burnham and Robert Spier

GSoC Program Administrators

---

You can find the original copy of this text in [GSoC 2023 Mentor and Org Admin email archive](https://docs.google.com/document/d/1K3J323__RkqGJGFmFCQ-BJMt-S9Ct0cRojD2KgMlpAk/edit?pli=1#)

**Comment by nate-double-u** [comment]
There is a bug with the ignore feature, essentially clicking ignore then hides a proposal from everyone, not just the person clicking ignore.

We're going through to see if we can clean things up, and will be reaching out to GSoC for help if we need. My understanding is that all the proposals are still there, they just aren't visible to everyone.

**Reply by aliok** [reply]
An update here...

We went through all proposals and removed them from the ignore list. All proposals should be visible again.

We ask all mentors to not use the "ignore" feature. Thanks!

**Comment by nate-double-u** [comment]
The GSoC administrators also shared this attached to their email above:
* [Being a Great Google Summer of Code Mentor](https://youtu.be/3J_eBuYxcyg)

**Comment by aliok** [comment]
/cc @Abhishek-kumar09 @agardnerit @adityasaky @aliok @aslom @Ankurk99 @RealAnna @aavarghese @kranurag7 @bwplotka @daemon1024 @slashben @bradmccoydev @everettraven @camilamacedo86 @xu-chaojie @creydr @chrisohaver @craigbox @Somefive @TheSpiritXIII @DavidPHirsch @evankanderson @bacherfl @suprjinx @wu-hanqing @hydai @jasondellaluce @jefree-cat @Ericson2314 @joe-kimmel-vmw @kannon92 @kkourt @leecalcote @lionelvillard @mheon @matzew @mnm678 @macxamin @beeme1mr @michi-covalent @natalieparellano @ppatierno @haircommander @pierDipi @barnettZQG @rashmigottipati @ReToCode @SantiagoTorres @saschagrunert

(folks registered as mentors on GSoC platform for 2023)

**Reply by aliok** [reply]
/cc @skonto @Kavinjsir @thschue @varshaprasad96 @fourierr @ilixiaocui @q82419 @yongtang @yurishkuro

(remaining folks registered as mentors on GSoC platform for 2023)

**Comment by nate-double-u** [comment]
Hello everyone,

The deadline for GSoC ranked selections is **April 27, 2023**; however, we (the CNCF GSoC admin team) will need time to do those rankings, so we are asking mentors to provide their selections for ranking by **April 19, 2023**. This is to give us time to review and reach out to mentors in case of any conflicts. We aim to have these ranked and ready for Google by **April 25, 2023**.

We are aware that this deadline falls in the middle of KubeCon EU, so given the GSoC timeline, we ask that you **choose your proposals as soon as possible**. 

Platform link for the proposals: https://summerofcode.withgoogle.com/organizations/cncf/programs/2023/proposals/details

Please click the “Want to Mentor” button to indicate your interest in any particular proposal.

When you are selecting proposals, keep these in mind: Proposals should
* be well  written
* demonstrate an understanding of issues/project reqs
* show that the applicant has the skills to be successful
* show that they have a good understanding of the problem and have an idea of how to tackle it over the course of the program.

Bonus points for creativity/innovative solutions – let us know if a proposal is particularly interesting.

**Note**, If there are multiple proposals for  a project idea, the org admins will reach out to the projects (mentors) to see which they are most interested in working on. There can only be one mentee per project idea.

**Reply by jasondellaluce** [reply]
Hi @nate-double-u, thanks for all the details.

> Note, If there are multiple proposals for a project idea, the org admins will reach out to the projects (mentors) to see which they are most interested in working on.

This is already my case. When I'll get to a decision, should I express it through the “Want to Mentor” button, or do the org admins prefer me to reach out to them privately?

**Reply by aliok** [reply]
Hello @jasondellaluce 

Yes, please express it through the “Want to Mentor”.

> Note, If there are multiple proposals for a project idea, the org admins will reach out to the projects (mentors) to see which they are most interested in working on. There can only be one mentee per project idea.

We wrote this together with @nate-double-u and the situation that we tried to describe was when there are multiple proposals with "Want to Mentor" pressed for the same project idea.

Thanks for the feedback!

**Reply by ilixiaocui** [reply]
> Hi @nate-double-u, thanks for all the details.
> 
> > Note, If there are multiple proposals for a project idea, the org admins will reach out to the projects (mentors) to see which they are most interested in working on.
> 
> This is already my case. When I'll get to a decision, should I express it through the “Want to Mentor” button, or do the org admins prefer me to reach out to them privately?


Hi @nate-double-u, this is already my case. Do I just need to click the "Want to Mentor" button on one of the proposals，and 
no need to do anything with other proposals?

**Reply by aliok** [reply]
hi @ilixiaocui 

> just need to click the "Want to Mentor" button on one of the proposals

This would work fine. Thanks for reaching out and making sure!

**Reply by ilixiaocui** [reply]
> hi @ilixiaocui
> 
> > just need to click the "Want to Mentor" button on one of the proposals
> 
> This would work fine. Thanks for reaching out and making sure!

Thanks @nate-double-u

**Comment by kannon92** [comment]
Hello,

What is considered the org admin for this group?  

One of the mentors of the Armada project didn’t see the email asking him to join CNCF GSOC so he’d like to get another invite.

**Reply by aliok** [reply]
Hi @kannon92 

Thanks for reaching out.

I assume you're asking the invite for Geoffrey Wilson, listed in https://github.com/cncf/mentoring/blob/main/summerofcode/2023.md#build-interfaces-around-postgres-for-armada.

I don't see Geoffrey in GSoC system as registered. When I want to send a new invite to the email address (geoff AT gr-oss DOT com) that's listed in the page above, GSoC system says "Invite for this user to this organization already exists".

We've had experienced some bugs in GSoC system this year. Maybe this is one of them. I will reach out to GSoC program managers at Google. In the mean time, I can send an invite to another email address of Geoffrey's if you provide me one.

**Reply by aliok** [reply]
@kannon92 @suprjinx

Google program managers told me that they resent the invitation email now. Could you please check?

**Reply by suprjinx** [reply]
Thank you -- I think the issue is that email address is incorrect @aliok -- I am geoff AT gr-oss DOT io

**Reply by aliok** [reply]
Hi @suprjinx I sent you a new invite now.
Could you please check? Once you accept the invite and login to the GSoC system, you need to accept agreements for 2023 program. To do that, go to https://summerofcode.withgoogle.com/organizations/cncf/programs and click on 2023. You will see the agreements there.

**Reply by suprjinx** [reply]
Thanks @aliok!

**Comment by nate-double-u** [comment]
Hi Everyone!

Google has informed us that we've had [14 projects selected for this year's program](https://github.com/cncf/mentoring/discussions/954)! 🚀

I want to thank everyone for putting in such thoughtful proposals.

For those with projects that weren't selected, I'd like to encourage you to submit them to the next terms of the CNCF LFX Mentorship program -- many of the GSoC project proposals can be copied almost one for one to a PR against an LFX term, especially if you've got strong submissions from contributors.

LFX Mentoring terms:
* [2023 Term 02](https://github.com/cncf/mentoring/tree/main/programs/lfx-mentorship/2023/02-Jun-Aug) - proposals due: Tue, May 9, 5:00 PM PDT
* [2023 Term 03](https://github.com/cncf/mentoring/tree/main/programs/lfx-mentorship/2023/03-Sep-Nov) - proposals due: Thur July 27, 5:00 PM PDT

**Reply by aliok** [reply]
Just a clarification here. These dates are deadlines for adding project ideas as mentors. Deadline for mentee applications is later.

(I was reached out by some mentors who got confused)

**Reply by nate-double-u** [reply]
That's correct--deadlines listed here are for mentors to propose ideas. The mentee applications period is opened after.
